### ISSUE TITLE
Inappropriate dependency in deb package

### ISSUE THREAD

**Issue by rosik** [tarantool/tarantool#5429]
Tarantool version: 2.4.2

```
$ apt show tarantool tarantool-common                                                                                                                                                   
Package: tarantool 
Version: 2.4.2.0.gcccd89701-1
Depends: ..., tarantool-common (>= 1.7.5.46)
```

Tarantool dependency on `tarantool-common` from is specified as `>= 1.7.5`, while it should be `== 2.4.2....`

The tarantoolctl binary belongs to the `tarantool-common` package, but all luarocks modules - to the `tarantool`. Version inconsistency causes problems.

**Comment by Totktonada** [comment]
The idea of `tarantool-common` and this weak dependency is to don't update init/service files (and restart the daemon) each time `tarantool` is updated. AFAIK, it is Debian policy to restart a service at update. AFAIR, we can skip restart by using DEB_DH_SYSTEMD* options. Maybe there were no such ability in the past or the idea was to restart tarantool anyway when the entire system is upgraded (and so tarantool-common is updated too).

In short: the weak dependency is right, but the minimal dependency should be bumped when given tarantoolctl version is not compatible with given tarantool version anymore.

**Comment by Totktonada** [comment]
The problem per so (reported by a user via @rosik to us):

```
/usr/bin/tarantoolctl:908: module 'luarocks.cfg' not found
```

So, we should depend on `tarantool-common` with tarantoolctl with luarocks-3 support for versions of tarantool, which ship luarocks-3.

**Comment by Mons** [comment]
Maybe better change shipment unit of luarocks module and move it to tarantoolctl package (common)?

**Comment by rosik** [comment]
I would also add that `tarantoolctl --help` reported 2.4.2, while in fact it was installed from 1.10 package, and it's misleading.

**Comment by Totktonada** [comment]
> Maybe better change shipment unit of luarocks module and move it to tarantoolctl package (common)?

Seems good on the first glance.

NB: If we'll do, we should correctly set which packages blocks each others to avoid conflicting by files.

**Comment by rosik** [comment]
The same issue just affected @Errora

![image](https://user-images.githubusercontent.com/2205188/101614801-70278480-3a1e-11eb-967c-583152b170b6.png)

The reason is the old tarantool-common 1.9, `apt upgrade` fixed it.

**Comment by DifferentialOrange** [comment]
> The idea of `tarantool-common` and this weak dependency is to don't update init/service files (and restart the daemon) each time `tarantool` is updated. AFAIK, it is Debian policy to restart a service at update. AFAIR, we can skip restart by using DEB_DH_SYSTEMD* options. Maybe there were no such ability in the past or the idea was to restart tarantool anyway when the entire system is upgraded (and so tarantool-common is updated too).
> 
> In short: the weak dependency is right, but the minimal dependency should be bumped when given tarantoolctl version is not compatible with given tarantool version anymore.

AFAIK, current deb packages skips restart so this weak dependency is not required anymore
https://github.com/tarantool/tarantool/blob/60f168184b8dfcafa6805a1e00a09012265b58db/debian/rules#L27

**Comment by Totktonada** [comment]
Seems so. #6400 is about that. We want to resolve current problems first, provide updates to Debian's repositories (simple update will not require much attention from Debian developers) and only after this rearrange the packages and propose the new packages structure to review into Debian.

**Comment by Totktonada** [comment]
This problem was fixed in tarantool 2.10.0.

| Package                | Requires                         | Conflicts                       |
| ---------------------- | -------------------------------- | ------------------------------- |
| tarantool-2.10.0       | tarantool-common >= 2.2.1        | tarantool-common (<< 2.2.1)[^1] |
| tarantool-common-2.10.0|                                  | tarantool-common (<< 2.2.1)     |
| tarantool-dev-2.10.0   | tarantool (exactly same version) | tarantool-common (<< 2.2.1)     |

[^1]: Strictly less than 2.2.1.

That's correct: `tarantool-2.10.0`, which expects luarocks 3, requires `tarantool-common` that contains luarocks 3 and conflicts with one that contains luarocks 2.

Let's look on latest 1.10 release:

| Package                  | Requires                         | Conflicts                  |
| ------------------------ | -------------------------------- | -------------------------- |
| tarantool-1.10.14        | tarantool-common >= 1.7.5.46     | tarantool-common (<< 1.6~) |
| tarantool-common-1.10.14 |                                  | tarantool-common (<< 1.6~) |
| tarantool-dev-1.10.14    | tarantool (exactly same version) | tarantool-common (<< 1.6~) |

`tarantool-1.10.14` requires `tarantool-common`, but it doesn't specify, whether it needs one with luarocks 2 (`tarantool-common << 2.2.1`) or one with luarocks 3 (`tarantool-common >= 2.2.1`). So both combinations are allowed.

It hits users on Debian Bullseye (11) and Ubuntu Jammy (22.04), because it contains `tarantool{,-common,-dev}-2.6.0`. For example: https://github.com/tarantool/setup-tarantool/issues/36 (starting from words 'tarantoolctl fatal error'). Previous Debian version (Debian Buster, 10) and previous LTS Ubuntu version (Ubuntu Focal, 20.04) offer tarantool 1.9, so there is no 1.10/2.* conflict (unless a user setups two tarantool.org repositories on its own).

Our `installer.sh` script has some protection against such problems, see https://github.com/tarantool/installer.sh/pull/11 (package origin pinning). But the script is not the only installation method, so on the first glance it looks meaningful to update the `control` file for tarantool 1.10.

However I don't know, to be honest, whether the fix would actually make users more happy. I guess that without pinning of the `installer.sh` kind, the packages will anyway updated to 2.6.0 at next `apt-get update`. So 1.10 user likely pins it on Debian Bullseye/Ubuntu Jammy. But if all the three packages are pinned, the user shouldn't meet the problem.

It is one more confirmation of #6400's point: the separate tarantool-common is error prone.
